### ISSUE TITLE
local variable 'a' referenced before assignment

### ISSUE THREAD

**Issue by Dpananos** [greta-dev/greta#280]
After installation of `greta`, `tensorflow`, and `tensorflow-probability`, I get the following error when running a simple example.

```
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  UnboundLocalError: local variable 'a' referenced before assignment 
```

Googling this error leads to a GitHub issue [here](https://github.com/rstudio/keras/issues/693) which suggests downgrading to `tensorflow=1.12`, but doing so throws an error when trying to run the example below saying that greta needs the most recent tensorflow release.

```r
x = iris$Petal.Length
y = iris$Sepal.Length

library(greta)

int<-normal(0,1)
coef<-normal(0,1)
sd<-lognormal(0,1)
mean <- int+coef*x
distribution(y) <- normal(mean,sd)
m <- model(int,coef,sd)
draws<- mcmc(m,n_samples = 1000, chains = 4)

```

Here is the contents of the `r-tensorflow` environment
```

# packages in environment at /Users/demetri/anaconda3/envs/r-tensorflow:
#
# Name                    Version                   Build  Channel
absl-py                   0.7.1                    pypi_0    pypi
asn1crypto                0.24.0                py37_1003    conda-forge
astor                     0.7.1                    pypi_0    pypi
ca-certificates           2019.3.9             hecc5488_0    conda-forge
certifi                   2019.3.9                 py37_0    conda-forge
cffi                      1.12.3           py37hccf1714_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
cryptography              2.6.1            py37h212c5bf_0    conda-forge
freetype                  2.10.0               h24853df_0    conda-forge
gast                      0.2.2                    pypi_0    pypi
grpcio                    1.20.1                   pypi_0    pypi
h5py                      2.9.0                    pypi_0    pypi
hdf5                      1.10.4          nompi_h0cbb7df_1106    conda-forge
idna                      2.8                   py37_1000    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
keras                     2.2.4                    pypi_0    pypi
keras-applications        1.0.7                    pypi_0    pypi
keras-preprocessing       1.0.9                    pypi_0    pypi
libblas                   3.8.0                8_openblas    conda-forge
libcblas                  3.8.0                8_openblas    conda-forge
libcxx                    4.0.1                hcfea43d_1
libcxxabi                 4.0.1                hcfea43d_1
libedit                   3.1.20181209         hb402a30_0
libffi                    3.2.1                h475c297_4
libgfortran               3.0.1                         0    conda-forge
liblapack                 3.8.0                8_openblas    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libtiff                   4.0.10            h79f4b77_1001    conda-forge
markdown                  3.1                      pypi_0    pypi
mock                      3.0.4                    pypi_0    pypi
ncurses                   6.1                  h0a44026_1
numpy                     1.16.3           py37hdf140aa_0    conda-forge
olefile                   0.46                       py_0    conda-forge
openblas                  0.3.6                hd44dcd8_1    conda-forge
openssl                   1.1.1b               h01d97ff_2    conda-forge
pillow                    6.0.0            py37h7095ceb_0    conda-forge
pip                       19.1                     py37_0
protobuf                  3.7.1                    pypi_0    pypi
pycparser                 2.19                     py37_1    conda-forge
pyopenssl                 19.0.0                   py37_0    conda-forge
pysocks                   1.6.8                 py37_1002    conda-forge
python                    3.7.3                h359304d_0
pyyaml                    5.1              py37h1de35cc_0    conda-forge
readline                  7.0                  h1de35cc_5
requests                  2.21.0                py37_1000    conda-forge
scipy                     1.2.1            py37hbd7caa9_1    conda-forge
setuptools                41.0.1                   pypi_0    pypi
six                       1.12.0                   pypi_0    pypi
sqlite                    3.28.0               ha441bb4_0
tensorboard               1.13.1                   pypi_0    pypi
tensorflow                1.13.1                   pypi_0    pypi
tensorflow-estimator      1.13.0                   pypi_0    pypi
tensorflow-hub            0.4.0                    pypi_0    pypi
tensorflow-probability    0.6.0                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
tk                        8.6.9             ha441bb4_1001    conda-forge
urllib3                   1.24.2                   py37_0    conda-forge
werkzeug                  0.15.2                   pypi_0    pypi
wheel                     0.33.1                   pypi_0    pypi
xz                        5.2.4                h1de35cc_4
yaml                      0.1.7             h1de35cc_1001    conda-forge
zlib                      1.2.11               h1de35cc_3


```

**Comment by Voltemand** [comment]
Hi, 

Thanks for the bug report!

Can you please post your session info - run 

```
sessionInfo()
```

Best!

**Comment by boralprophecy** [comment]
Got same error.

My session info is : 

```
> sessionInfo()
R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] greta_0.3.0.9002

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1             pillar_1.3.1           compiler_3.5.3         plyr_1.8.4             base64enc_0.1-3        prettyunits_1.0.2      tools_3.5.3            progress_1.2.0        
 [9] digest_0.6.18          jsonlite_1.6           tibble_2.1.1           gtable_0.2.0           lattice_0.20-38        pkgconfig_2.0.2        rlang_0.3.4            Matrix_1.2-15         
[17] rstudioapi_0.10        parallel_3.5.3         coda_0.19-2            dplyr_0.8.0.1          globals_0.12.4         hms_0.4.2              grid_3.5.3             tidyselect_0.2.5      
[25] reticulate_1.12.0-9000 glue_1.3.1             listenv_0.7.0          R6_2.4.0               whisker_0.3-2          ggplot2_3.1.0          purrr_0.3.2            magrittr_1.5          
[33] tfruns_1.4             scales_1.0.0           codetools_0.2-16       assertthat_0.2.1       future_1.12.0          colorspace_1.4-0       tensorflow_1.13.1.9000 lazyeval_0.2.1        
[41] munsell_0.5.0          crayon_1.3.4          
> 
```

**Comment by Dpananos** [comment]
My SessionInfo()
```

R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] greta_0.3.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1        rstudioapi_0.10   whisker_0.3-2     magrittr_1.5     
 [5] hms_0.4.2         progress_1.2.0    lattice_0.20-38   R6_2.4.0         
 [9] rlang_0.3.4       globals_0.12.4    tools_3.5.3       parallel_3.5.3   
[13] grid_3.5.3        coda_0.19-2       tfruns_1.4        digest_0.6.18    
[17] assertthat_0.2.1  crayon_1.3.4      tensorflow_1.13.1 Matrix_1.2-15    
[21] base64enc_0.1-3   codetools_0.2-16  compiler_3.5.3    prettyunits_1.0.2
[25] reticulate_1.12   jsonlite_1.6      future_1.12.0     listenv_0.7.0    
[29] pkgconfig_2.0.2  
```

**Comment by Voltemand** [comment]
See my comment in #282 

Best!

**Comment by goldingn** [comment]
this should be fixed in greta 0.3.1, but please reopen if you get the same error in that version
